src/Service/Resolvers/Sql Query Structures/SqlUpsertQueryStructure.cs

```diff
@@ -121,9 +121,9 @@ public ColumnDefinition GetColumnDefinition(string columnName)
                     MetadataProvider.TryGetBackingColumn(EntityName, param.Key, out string? backingColumn);
                     // Create Parameter and map it to column for downstream logic to utilize.
                     string paramIdentifier;
-                    if (param.Value != null)
+                    if (param.Value is not null)
                     {
-                        paramIdentifier = MakeParamWithValue(GetParamAsColumnSystemType(param.Value.ToString()!, backingColumn!));
+                        paramIdentifier = MakeParamWithValue(GetParamAsSystemType(param.Value.ToString()!, backingColumn!, GetColumnSystemType(backingColumn!)));
```

Comment:
We should be using the mapped column to provide the error message - yet use the backingColumn to obtain the type

Reply:
Thanks for the suggestion! I've made the changes to return the mapped column names in the error message